lib/membrane_rtc_engine/endpoints/webrtc/rtp_connection_allocator.ex

```diff
@@ -18,10 +18,27 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.RTPConnectionAllocator do
            negotiable?: boolean()
          }
 
+  # Type describing a status of RTPConnectionAllocator's prober part.
+
+  # - `:increase_estimation` - actively probing slightly over current estimation in order to increase it
+  # - `:maintain_estmation` - only sending enough probes to make sure that the estimation doesn't fall
+  #   below the level of allocated bandwidth
+  # - `:allowed_bandwidth_deficiency` - we have allocated more bandwidth than the estimation, but the estimation
+  #  is increasing. It's only possible to reach this state though allocations from non-negotiable Track
+  #  Receivers and through new Track Receiver registering
+  # - `:disallowed_bandwidth_deficiency` - we have allocated more bandwidth than estimated. This state can be reached
+  #  when we're in `:allowed_bandwidth_deficiency` and estimation decreased, or if the estimation decreases below
+  #  the level of total allocation
+  @typep prober_status_t() ::
+           :increase_estimation
+           | :maintain_estimation
+           | :allowed_bandwidth_deficiency
+           | :disallowed_bandwidth_deficiency
```

Comment:
I think that `allowed_overuse` or `allowed_overshot` will more precisely indicate what we mean here